File: src/main/rest/repository/BriefingRepository.js
import Briefing from "../model/Briefing";


const BriefingRepository = {

    selectByPurposeId : (txn, purposeId) => {
        return new Promise((resolve, reject) => {
            txn.executeSql(
                'select purpose_id, goal_id, create_at, score from briefing where purpose_id = ?',
                [purposeId],
                (tx, res) => {
                    let data = [];
                    for(let i =0; i < res.rows.length; i++){
                        const row = res.rows.item(i);
                        data.push(new Briefing(row.purpose_id, row.goal_id, row.create_date, row.score))
                    }

                    resolve(data);
                },
                (error) => {
                    reject(error);
                }
            )
        })
    },

    insert : (txn, purposeId, goalId) => {
        return new Promise((resolve, reject) => {
            txn.executeSql(
                'insert into briefing(purpose_id, goal_id, create_at, score) values(?, ?, datetime(\'now\'), 0',
                [purposeId, goalId],
                (tx, res) => {
                    resolve();
                },
                (error) => {
                    reject(error);
                }
            )
        })
    }

}

export default BriefingRepository;

File: android/app/src/main/java/com/downfall/caterplanner/rest/repository/BriefingRepository.java
package com.downfall.caterplanner.rest.repository;

import android.database.Cursor;

import com.downfall.caterplanner.rest.model.Briefing;
import com.downfall.caterplanner.rest.db.SQLiteManager;
import com.downfall.caterplanner.util.DateUtil;

import java.text.ParseException;
import java.util.ArrayList;
import java.util.List;

public class BriefingRepository extends BaseRepository{


    public List<Briefing> selectByHeaderId(long headerId) throws ParseException {
        final String sql =
                "select header_id, perfrom_id, create_at, score " +
                        "from briefing " +
                        "where header_id = ?";
        Cursor c = db.rawQuery(sql, new String[]{String.valueOf(headerId)});
        List<Briefing> briefings = new ArrayList<>();
        while(c.moveToNext()){
            briefings.add(
                    Briefing.builder()
                            .headerId(c.getLong(1))
                            .performId(c.getInt(3))
                            .createAt(DateUtil.parseToDateTime(c.getString(4)))
                            .score(c.getInt(5))
                            .build()
            );
        }
        return briefings;
    }


    public void insert(long headerId, int performId){
        final String sql =
                "insert into briefing values(?, ?, datetime(\'now\'), 0)";
        db.execSQL(sql, new String[]{String.valueOf(headerId), String.valueOf(performId)});
    }


}
